### PR TITLE
Updating the vtk-sitk conversion code.

### PR DESCRIPTION
**Pull request by zivy**
Removed debugging code, streamlined the remaining code and added type hints and documentation.

### CODE REVIEW

File: SimpleITK/utilities/vtk.py [hunk 1]
```diff
@@ -64,63 +63,52 @@ def sitk2vtk(img, debugOn=False):
...
+    # Create VTK image and set its metadata
+    vtk_image = vtk.vtkImageData()
     vtk_image.SetDimensions(size)
     vtk_image.SetSpacing(spacing)
     vtk_image.SetOrigin(origin)
     vtk_image.SetExtent(0, size[0] - 1, 0, size[1] - 1, 0, size[2] - 1)
-
     if vtk.vtkVersion.GetVTKMajorVersion() < 9:
-        print("Warning: VTK version <9.  No direction matrix.")
+        logging.warning(
```
Comment (blowekamp) [inline SimpleITK/utilities/vtk.py:73]: The root logger should not be used. A package specific logger should be created to allow control over logging at the package level.

File: SimpleITK/utilities/vtk.py [hunk 2]
```diff
@@ -64,63 +63,52 @@ def sitk2vtk(img, debugOn=False):
...
+    sitk_image.SetOrigin(image.GetOrigin())
+    # By default, direction is identity.
 
-        print("numpy type:", npdata.dtype)
-        print("numpy shape:", npdata.shape)
+    if vtk.vtkVersion.GetVTKMajorVersion() >= 9:
+        # Copy the direction matrix into a list
+        dir_mat = image.GetDirectionMatrix()
+        direction = [0] * 9
+        dir_mat.DeepCopy(direction, dir_mat)
```
Comment (blowekamp) [inline SimpleITK/utilities/vtk.py:111]: Not sure why this change was made here too.

**Review by blowekamp — COMMENTED** [review]
These changes seem more that just documentation/debug related. The testing needs to be improved.

Also this is not the correct approach to logging.

File: SimpleITK/utilities/vtk.py (hunk 2 above)
Comment (zivy) [reply SimpleITK/utilities/vtk.py:111]: Following the approach [taken in Slicer](https://github.com/Slicer/Slicer/blob/e53e8af9c4a0b60adee28b5eca5fc1b5ff2da9ea/Base/Python/slicer/util.py#L1129-L1130). Code is more concise.

File: SimpleITK/utilities/vtk.py (hunk 1 above)
Comment (zivy) [reply SimpleITK/utilities/vtk.py]: Fixed.

Comment (blowekamp) [reply SimpleITK/utilities/vtk.py]: The module specific logger is an OK approach too.

**Review by blowekamp — APPROVED** [review]
